src/coord/src/coord.rs

```diff
@@ -164,101 +159,52 @@ impl<C> Coordinator<C>
+        let ts = SystemTime::now()
+            .duration_since(SystemTime::UNIX_EPOCH)
+            .expect("failed to get millis since epoch")
+            .as_millis()
+            .try_into()
+            .expect("current time did not fit into u64");
 
-        if let Some(granularity) = config.logging_granularity {
-            broadcast(
-                &mut broadcast_tx,
-                SequencedCommand::EnableLogging(LoggingConfig {
-                    granularity_ns: granularity.as_nanos(),
-                    active_logs: BUILTINS
-                        .logs()
-                        .map(|src| (src.variant.clone(), src.index_id))
-                        .collect(),
-                }),
-            )
-            .await;
+        if ts < self.read_lower_bound {
+            self.read_lower_bound
+        } else {
+            ts
         }
+    }
 
-        let (persister, persistence_tx, persistence_path) =
-            if let Some(persistence_config) = config.persistence {
-                let (persistence_tx, persistence_rx) = config.switchboard.mpsc();
-                broadcast(
-                    &mut broadcast_tx,
-                    SequencedCommand::EnablePersistence(persistence_tx.clone()),
-                )
-                .await;
-                let path = persistence_config.path.clone();
-                (
-                    Some(Persister::new(persistence_rx, persistence_config)),
-                    Some(
-                        persistence_tx
-                            .connect()
-                            .await
-                            .expect("failed to connect persistence tx"),
-                    ),
-                    Some(path),
-                )
-            } else {
-                (None, None, None)
-            };
-
-        let mut coord = Self {
-            switchboard: config.switchboard,
-            cmd_rx: Some(config.cmd_rx),
-            broadcast_tx,
-            num_timely_workers: config.num_timely_workers,
-            optimizer: Default::default(),
-            catalog,
-            symbiosis,
-            indexes: ArrangementFrontiers::default(),
-            since_updates: Vec::new(),
-            active_tails: HashMap::new(),
-            logging_granularity: config
-                .logging_granularity
-                .and_then(|c| c.as_millis().try_into().ok()),
-            timestamp_config: config.timestamp,
-            logical_compaction_window_ms,
-            feedback_rx: Some(rx),
-            persister,
-            persistence_tx,
-            persistence_path,
-            closed_up_to: 1,
-            read_lower_bound: 1,
-            last_op_was_read: false,
-            need_advance: true,
-            transient_id_counter: 1,
-        };
-
-        let catalog_entries: Vec<_> = coord
+    /// Initializes coordinator state based on the contained catalog. Must be
+    /// called after creating the coordinator and before calling serve.
```

Comment:
consider ending the comment with "... before calling Coordinator::serve."

mildly confusing because this file has two functions named `serve`

Reply:
Ack, done.